app/controllers/items_controller.rb

```diff
@@ -0,0 +1,44 @@
+class ItemsController < ApplicationController
+  def index
+    if params[:merchant_id].nil?
+      @items = Item.all
+    else
+      @merchant = Merchant.find(params[:merchant_id])
+
+      @items = @merchant.items
+    end
+  end
+
+  def show
+    @item = Item.find(params[:id])
+    @merchant = Merchant.find(@item.merchant_id)
```

Comment:
since @ item already has a merchant_id, when you call @ item.merchant, it will return that merchant object. That being said, it's not necessary to define @ merchant in your controller since you can just call @ item.merchant in your view, but I'm fine with it being here too. However, if you left it, I'd change line 14 to be: 
```ruby
@ merchant = @item.merchant 
```